### DISCUSSION TITLE
How do you use the Extension Creator?

### DISCUSSION THREAD

**Discussion by McStebb** [dialogic-godot/dialogic#1820] [Q&A] [answered]
Hi there.

Sorry for the rather open-ended question... but I can't find any documentation on how this is used:
![image](https://github.com/coppolaemilio/dialogic/assets/46116575/da0eb959-7e09-4c0a-973d-c2d0c2d54c96)

I want to create a custom event that spawns a graphic button which, when click, takes you to a different timeline. Same function as a normal choice button, but with a graphic attached to it.

Right now I got it to work using `call_node` and it works perfectly fine, but when I saw the Extension Creator I got the feeling that there was a better way to do it using a custom event.

Is there any documentation on how to use custom events in 2.0? I probably just missed it.

**Comment by Jowan-Spooner** [comment, marked as answer]
Try this: [Creating Extensions](https://github.com/dialogic-godot/documentation/blob/main/Documentation/Creating_Extensions.md)
As all the 2.0 documentation, it's not perfect, but it's the best we've got. If anything doesn't work or you have further questions feel free to ask.

**Comment by McStebb** [comment]
I was able to piece together what to do.  Thanks for the help!
